sfr/subchannel/multiple_SCM_assemblies/7assemblies/abr_7assemblies.i

```diff
@@ -0,0 +1,321 @@
+  []
+
+  [inlet_interwrapper]
+    type = ParsedGenerateSideset
+    input = extrude # inlet_interwall
+    combinatorial_geometry = 'abs(z) < 1e-6'
+    included_subdomains = '1006 1003'
+    normal = '0 0 -1'
+    new_sideset_name = inlet_interwrapper
+  []
+
+  [inlet_porous_flow_hfd]
+    type = ParsedGenerateSideset
+    input = inlet_interwrapper
+    combinatorial_geometry = 'abs(z) < 1e-6'
+    included_subdomains = 13
+    normal = '0 0 -1'
+    new_sideset_name = inlet_porous_flow_01
+  []
+
+  [inlet_porous_flow_p]
+    type = ParsedGenerateSideset
+    input = inlet_porous_flow_hfd
+    combinatorial_geometry = 'abs(z) < 1e-6'
+    included_subdomains = 14
+    normal = '0 0 -1'
+    new_sideset_name = inlet_porous_flow_02
+  []
+
+  [inlet_porous_flow_d1]
+    type = ParsedGenerateSideset
+    input = inlet_porous_flow_p
+    combinatorial_geometry = 'abs(z) < 1e-6'
+    included_subdomains = 15
+    normal = '0 0 -1'
+    new_sideset_name = inlet_porous_flow_03
+  []
+
+  [inlet_porous_flow_d2]
+    type = ParsedGenerateSideset
+    input = inlet_porous_flow_d1
+    combinatorial_geometry = 'abs(z) < 1e-6'
+    included_subdomains = 16
+    normal = '0 0 -1'
+    new_sideset_name = inlet_porous_flow_04
+  []
+
+  [inlet_porous_flow_k011]
+    type = ParsedGenerateSideset
+    input = inlet_porous_flow_d2
+    combinatorial_geometry = 'abs(z) < 1e-6'
+    included_subdomains = 17
+    normal = '0 0 -1'
+    new_sideset_name = inlet_porous_flow_05
+  []
+
+  [inlet_porous_flow_x402]
+    type = ParsedGenerateSideset
+    input = inlet_porous_flow_k011
+    combinatorial_geometry = 'abs(z) < 1e-6'
+    included_subdomains = 18
+    normal = '0 0 -1'
+    new_sideset_name = inlet_porous_flow_06
+  []
+
+  [inlet_central_assembly]
+    type = ParsedGenerateSideset
+    input = inlet_porous_flow_x402
+    combinatorial_geometry = 'abs(z) < 1e-6'
+    included_subdomains = '12'
+    normal = '0 0 -1'
+    new_sideset_name = inlet_porous_flow_00
+  []
+
+  [outlet_interwrapper]
+    type = ParsedGenerateSideset
+    input = inlet_central_assembly
+    included_subdomains = '1006 1003'
+    combinatorial_geometry = 'abs(z - ${fparse height}) < 1e-6'
+    normal = '0 0 1'
+    new_sideset_name = outlet_interwrapper
+  []
+
+  [outlet_porous_flow]
+    type = ParsedGenerateSideset
+    input = outlet_interwrapper
+    included_subdomains = '13 14 15 16 17 18'
+    combinatorial_geometry = 'abs(z - ${fparse height}) < 1e-6'
+    normal = '0 0 1'
+    new_sideset_name = outlet_porous_flow
+  []
+
+  [outlet_central_assembly]
+    type = ParsedGenerateSideset
+    input = outlet_porous_flow
+    included_subdomains = '12'
+    combinatorial_geometry = 'abs(z - ${fparse height}) < 1e-6'
+    normal = '0 0 1'
+    new_sideset_name = outlet_porous_flow_00
+  []
+
+  [rename]
+    type = RenameBlockGenerator
+    input = outlet_central_assembly
+    old_block = '1003     1006          12'
+    new_block = 'wall inter_wrapper porous_flow_00'
+  []
```

Comment:
just need to move this as early as possible, before the sideset generation at least